pubsub/gossipsub/gossipsub-v1.2.md

```diff
@@ -0,0 +1,105 @@
+v1.1](https://github.com/libp2p/specs/blob/master/pubsub/gossipsub/gossipsub-v1.1.md)
+protocol.
+
+The proposed extensions are backwards-compatible and aim to add utility control messages for 
+better network efficiency (delivery latency and total throughput) tuning of gossip protocol 
+implementations 
+
+In more specific terms, new control messages are introduced: `PING` and `PONG`. 
+They are primarily intended for collecting statistics to estimate peer connection 
+latency and bandwidth and adjusting message broadcasting strategy accordingly.
+
+# Specification
+
+## Protocol Id
+
+Nodes that support this Gossipsub extension should additionally advertise the
+version number `1.2.0`. Gossipsub nodes can advertise their own protocol-id
+prefix, by default this is `meshsub` giving the default protocol id:
+- `/meshsub/1.2.0`
+
+## Parameters
+
+This section lists the configuration parameters that needs to agreed on across clients to avoid
+peer penalizations
+
+| Parameter         | Description                                                            | Reasonable Default |
+|-------------------|------------------------------------------------------------------------|--------------------|
+| `min_ping_period` | minimal delay between sending next `PING` and previous received `PONG` | 1 second (?)       |
+| `max_ping_rate`   | (better alternative) maximum number of `PING` messages per minute      | 60                 |
+ * [either `min_ping_period` or `max_ping_rate`]
+
+## PING/PONG Messages
+
+### Sender
+
+Sender MAY send `PING` message (either standalone or piggy tailing) at any time. Sender is free choose 
+any `id` however it might be reasonable to select unique numbers across the session. 
+
+Sender MUST NOT send next `PING` message earlier than `min_ping_period` after receiving the previous 
+`PONG` message (if any). [or alternatively respect `max_ping_rate`]
+
+An implementation is free to choose a reasonable timeout (e.g. 60 seconds) for waiting for a corresponding 
+`PONG` message. However it is worth considering that a `PONG` message might be 
+[... blocked by another message which is already on the wire] and not setting the timeout too small.
+If a `PONG` timeout occurs that SHOULD be treated as protocol malfunction and the peer connection 
+SHOULD be closed  
+
+### Responder
+
+Upon receiving a `PING` message the responder
+- MUST respond with a `PONG` message with the same `id` value
+- SHOULD check if the `min_ping_period` is not violated (otherwise the peer MAY be penalized)
+  [or alternatively `max_ping_rate` is not voilated]
+- SHOULD make his best to respond back with a corresponding `PONG` as soon as possible. That means:
+  - SHOULD NOT queue the message for later send (e.g. on heartbeat or piggy tailing for a next publish message)
+  - if there are messages queued for sending to the peer that queue SHOULD be disregarded and the 
+    the `PONG` message should be flushed immediately
+
+### Usage scenarios
```

Comment:
Can also talk about the connection "keep alive" (or keep warm I guess)

Reply:
Oh, that's actually too weird use case from my perspective. 
I'm not sure that dirty hack is worth mentioning in libp2p spec 😆